emready/tests/test_protocol_sharpening.py

```diff
@@ -0,0 +1,114 @@
+
+logger = logging.getLogger(__name__)
+from pyworkflow.tests import BaseTest, setupTestProject, DataSet
+from pwem.protocols import ProtImportVolumes, ProtImportMask
+from pyworkflow.utils import magentaStr
+
+from ..protocols import ProtEMReadySharppening
+
+
+class TestEMReadySharpening(BaseTest):
+    @classmethod
+    def setUpClass(cls):
+        setupTestProject(cls)
+        cls.dataSet = DataSet.getDataSet('xmipp_tutorial')
+
+        # Imports
+        logger.info(magentaStr("\n==> Importing data - Input data"))
+        new = cls.proj.newProtocol  # short notation
+        launch = cls.proj.launchProtocol
+
+        # Volumes
+        logger.info(magentaStr("\nImporting Volumes:"))
+
+        # Volume 1
+        pImpVolume = new(ProtImportVolumes, samplingRate=1,
+                         filesPath=cls.dataSet.getFile('vol2'))
+        launch(pImpVolume, wait=True)
+        cls.inputVol = pImpVolume.outputVolume
+
+        # Volume 2
+        pImpVolume2 = new(ProtImportVolumes, samplingRate=1,
+                          filesPath=cls.dataSet.getFile('vol1'))
+
+        launch(pImpVolume2, wait=True)
+        cls.inputVol2 = pImpVolume2.outputVolume
+
+        # This can be removed if not used
+        # References
+        # logger.info(magentaStr("\nImporting References:"))
+        # pImpRef = new(ProtImportVolumes, samplingRate=1,
+        #               filesPath=cls.dataSet.getFile('vol3'))
+        # launch(pImpRef, wait=True)
+        # #   reference.vol
+        # cls.inputRef = pImpRef.outputVolume
+        # pImpRef2 = new(ProtImportVolumes, samplingRate=1,
+        #                filesPath=cls.dataSet.getFile('vol1'))
+        # launch(pImpRef2, wait=True)
+        # cls.inputRef2 = pImpRef2.outputVolume
+        #
+        # # Masks
+        # logger.info(magentaStr("\nImporting Mask:"))
+        # pImpMask = new(ProtImportMask,
+        #                maskPath=cls.dataSet.getFile('mask3d'),
+        #                samplingRate=1)
+        # launch(pImpMask, wait=True)
+        # cls.mask = pImpMask.outputMask
+
+    def testSharpening(self):
+        """ Check that an output was generated and the condition is valid.
+            In addition, returns the size of the set.
+        """
+        logger.info(magentaStr("\n==> Testing EM ready:"))
+
+        def launchTest(label, vol,):
+            logger.info(magentaStr("\nTest %s:" % label))
+            emReadyProtocol = self.proj.newProtocol(ProtEMReadySharppening,
+                                              objLabel='EMReady %s' % label,
+                                              in_vol=vol)
+            self.proj.launchProtocol(emReadyProtocol)
+
+            outputVol = getattr(emReadyProtocol, EMReadyOutputs.sharpenedVolume.name)
+            self.assertIsNotNone(outputVol,
+                                 "Sharpened volume is None for %s test." % label)
+
+            self.assertEqual(self.inputVol.getDim(),
```

Comment:
The sampling rate of volume will be interpolated to 1.0 angstrom and therefore the size of volume might be changed after applying EMReady. These two assertions should be removed.